xbmc/music/tags/TagLoaderTagLib.cpp

```diff
@@ -302,6 +326,27 @@ bool CTagLoaderTagLib::ParseTag(ID3v2::Tag *id3v2, MUSIC_INFO::EmbeddedArt *art,
         else if (g_advancedSettings.m_logLevel == LOG_LEVEL_MAX)
           CLog::Log(LOGDEBUG, "unrecognized user text tag detected: TXXX:%s", frame->description().toCString(true));
       }
+    else if (it->first == "TIPL")
+      // Loop through and process the involved people list 
+      // For example Arranger, Engineer, Producer, DJMixer or Mixer
+      // In fieldlist every odd field is a function, and every even is an artist or a comma delimited list of artists.
+      for (ID3v2::FrameList::ConstIterator ip = it->second.begin(); ip != it->second.end(); ++ip)
+      {
+        ID3v2::TextIdentificationFrame *tiplframe = reinterpret_cast<ID3v2::TextIdentificationFrame*> (*ip);
+        if (!tiplframe) continue;
+        AddArtistRole(tag, StringListToVectorString(tiplframe->fieldList()));
+      }
+    else if (it->first == "TMCL")
```

Comment:
Beg to differ in the context, the pattern in that function is more readable without